### PR TITLE
Improve admin panel

### PR DESCRIPTION
**Pull request by frobs**
Required by #80 
imagen

### CODE REVIEW

File: views/templates/admin/configure_administration_panel.tpl [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<div class="panel ">
+  <div class="panel-heading">Panel de administracion</div>
+  <div class="panel-body">
+    <div class="container">
+      <div class="row">
+        <div class="col-md-6 col-sm-6">
+          <p><strong style="font-size: 1.2em;">Configurar Doofinder en mi tienda</strong></p>
+          <p style="font-size: 1.1em;">Accede al panel de Doofinder y descubre todas las funcionalidades para aumentar tus ventas.</p>
```
Comment (ivan-doofinder) [inline views/templates/admin/configure_administration_panel.tpl]: These texts must be translated

File: views/templates/admin/configure_administration_panel.tpl [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        </div>
+        <div class="col-md-6 col-sm-6 text-right">
+          <a href="https://admin.doofinder.com" class="btn btn-default btn-lg" target="_blank">
+            Ir al panel de Doofinder
```
Comment (ivan-doofinder) [inline views/templates/admin/configure_administration_panel.tpl]: Same

File: doofinder.php [hunk 3]
```diff
@@ -779,11 +796,17 @@ private function configureHookCommon($params = false)
...
-        $this->configureHookCommon($params);
-        if (Configuration::get('DF_ENABLED_V9')) {
-            return $this->displayScriptLiveLayer();
-        } else {
-            return $this->displayScriptV7();
+        $displayMobile = $this->getShowLayerMobileConfig();
+        $displayDesktop = $this->getShowLayerConfig();
+        $isMobile = Context::getContext()->isMobile();
+        if ((!empty($isMobile) && $displayMobile) ||
+        (empty($isMobile) && $displayDesktop)) {
```
Comment (ivan-doofinder) [inline doofinder.php]:
```suggestion
        if (getDisplayLayer())
```
Maybe it's something confusing. Better to create a function that contains the logic of whether to show the layer, and in the IF here just call that function.
In addition, the function can be used for other cases

File: doofinder.php [hunk 4]
```diff
@@ -342,12 +360,23 @@ protected function getConfigFormSearchLayer()
     protected function getConfigFormValuesSearchLayer()
     {
         $fields = [];
-
         $fields['DF_INSTALLATION_ID'] = Configuration::get('DF_INSTALLATION_ID');
+        $fields['DF_SHOW_LAYER'] = $this->getShowLayerConfig();
+        $fields['DF_SHOW_LAYER_MOBILE'] = $this->getShowLayerMobileConfig();
```
Comment (ivan-doofinder) [inline doofinder.php]:
```suggestion
        $fields['DF_SHOW_LAYER'] = Configuration::get('DF_SHOW_LAYER', true);
        $fields['DF_SHOW_LAYER_MOBILE'] = Configuration::get('DF_SHOW_LAYER_MOBILE', true);
```
Why call an intermediate function if you can get the values directly? (Same as for the other configuration values)

**Review by ivan-doofinder — CHANGES_REQUESTED** [review]

**Review by albertodoofinder — APPROVED** [review]

File: doofinder.php [hunk 5]
```diff
@@ -1681,4 +1690,14 @@ private function displayGeneralMsg($string, $type, $alert, $link = false)
...
     }
+
+    private function searchLayerMustBeInitialized()
+    {
+        $displayMobile = Configuration::get('DF_SHOW_LAYER_MOBILE', true);
+        $displayDesktop = Configuration::get('DF_SHOW_LAYER', true);
+        $isMobile = Context::getContext()->isMobile();
+
+        return (!empty($isMobile) && $displayMobile) ||
+        (empty($isMobile) && $displayDesktop);
```
Comment (ivan-doofinder) [inline doofinder.php]:
```suggestion
        return ($isMobile && $displayMobile) || (!$isMobile && $displayDesktop);
```
I think this is more readable

**Review by ivan-doofinder — CHANGES_REQUESTED** [review]

**Review by mdogo — APPROVED** [review]

**Review by ivan-doofinder — APPROVED** [review]

**Pull request merged**